### PR TITLE
SWE Playbook Meeting Changes

### PR DESCRIPTION
**Pull request by jmiller-rise8**

### CODE REVIEW

File: docs/practice-playbook.md [hunk 1]
```diff
@@ -1,150 +1,226 @@
 # Engineering Practice Playbook
 
-## Team Composition
+## Preamble
+This document contains our opinions on software developemnt. We understand that it is not always possible to hold to some of these standards. We trust each of our engineer's autonomy to respond to any given situtation.
```
Comment (Zojii) [inline docs/practice-playbook.md]: Add that this documents purpose is to produce repeatable high-value outcomes

File: docs/practice-playbook.md [hunk 2]
```diff
@@ -1,150 +1,226 @@
...
+    - engineering wage difference between one language to another
+    - tooling
+1. Is there support for the technology within the current continuous integration process?
+1. Can the technology be deployed to all environments?
+1. Can the technology be managed in all environments?
+1. Is the technology stack meeting security criteria and project constraints?
+1. Does the technology stack performance, reliability, and maintainability satisfy the product's requirements?
+1. Can the technology stack scale?
 
-Trust your team is making the best decisions with the information known at the moment, with or without your presence. You and your team have a common goal, there is more than one way to reach it.
```
Comment (Zojii) [inline docs/practice-playbook.md:36]: Trusting

File: docs/practice-playbook.md [hunk 3]
```diff
@@ -1,150 +1,226 @@
...
+## Discovery
+---
 
+### Build vs Buy Analysis
 
-### Collaboration
+Engineers evaluate already existing options to include commercial or government.
 
-Collaboration improves software delivery performance, organizational performance and job satisfaction[^1]. Working with someone to reach a common goal has many benefits to include:
+TODO: Separate into groups (Build & Purchase/Use)
```
Comment (jmiller-rise8) [inline docs/practice-playbook.md]: Update TODO 
> Todos should just be comments for expansion

https://github.com/ryneeverett/python-markdown-comments

File: docs/practice-playbook.md [hunk 4]
```diff
@@ -1,150 +1,226 @@
...
+## Building an MVP
+---
 
-### Courage
+Proof of Concept >> Prototype >> MVP
 
+### Proof of Concept
 
-### Mission Driven
+The goal of a Proof of Concept is for quick applied Technical Discovery to learn information to empower decision making. It can be comprised of pseudo-code, code-fragments, and/or diagrams that depict how the systems communicate. Outcome should be to validate & verify if feasible to accomplish, becomes the reference to for the prototype.
```
Comment (jmiller-rise8) [inline docs/practice-playbook.md]: grammatical issue

File: docs/practice-playbook.md [hunk 5]
```diff
@@ -1,150 +1,226 @@
...
 
+The whole document should be one or two pages long. We will write each ADR as if it is a conversation with a future developer. This requires good writing style, with full sentences organized into paragraphs. Bullets are acceptable only for visual style, not as an excuse for writing sentence fragments. (Bullets kill people, even PowerPoint bullets.)*
 
-### Technology Stack
 
-Selecting the right technology stack is important. Teams perform better when they have control of the tools and technologies. Here are a few things to consider when selecting tools and technologies:
 
+### Design Patterns and Principles
 
+When solving a problem or designing new features the engineer should consider patterns and principle before implementations.  Practice and patterns often produce performant, maintainable code.  There are times when you may need to deviate or create your own patterns.  In these cases you should document your the reason for your desing decision.
```
Comment (jmiller-rise8) [inline docs/practice-playbook.md]: grammatical error

File: docs/practice-playbook.md [hunk 6]
```diff
@@ -1,150 +1,226 @@
...
 
-1. Is training necessary?
-2. Are the skills and knowledge required common, or is the technology very niche?
-3. Is the technology mature enough to adopt?
-4. Are there any costs?
-5. Is there support for the technology within the current continuous integration process?
-6. Can the technology be deployed to all environments?
-7. Can the technology be managed in all environments?
+*Build list of common patterns and principles*
```
Comment (jmiller-rise8) [inline docs/practice-playbook.md]: Following section will be touched up by @Zojii

File: docs/practice-playbook.md [hunk 7]
```diff
@@ -1,150 +1,226 @@
...
 
-### Continuous Integration
+- SRP Single Responsibility principle  
+Make classes packages modules that solve one problem.
+- Open/Closed
 
-Continuous integration is a process that automates aspects of software development.
+- Liskov Substitution
 
+- Interfac Segregation
```
Comment (jmiller-rise8) [inline docs/practice-playbook.md]: typo

File: docs/practice-playbook.md [hunk 8]
```diff
@@ -1,150 +1,226 @@
...
 
+## Practices
+---
 
+Ceremonies
 
-* [https://martinfowler.com/articles/practical-test-pyramid.html](https://martinfowler.com/articles/practical-test-pyramid.html)
+### Story Point
 
+Engineers can help the team by helping to point stories. They can help estimate the amount or complexity of the work. Since engineers understand the work involved to fulfill a requirement,  they can ensure that stories are granular and right sized.
```
Comment (jmiller-rise8) [inline docs/practice-playbook.md:151]: spacing after comma

File: docs/practice-playbook.md [hunk 9]
```diff
@@ -1,150 +1,226 @@
...
 
-## Practices
+### Pair Programming
 
-Ceremonies
+Rotate pairs daily? after ticket? weekly?
```
Comment (jmiller-rise8) [inline docs/practice-playbook.md]: TODO

File: docs/practice-playbook.md [hunk 10]
```diff
@@ -199,49 +288,149 @@ Reviewers should _always_ feel free to leave comments expressing that something
...
-* Aspects of software design are seldom a pure style issue or just a personal preference**.** They are based on underlying principles and should be weighed on those principles, not simply by subjective opinion. Sometimes there are a few valid options. If the author can demonstrate (either through data or based on solid engineering principles) that several approaches are equally good, the reviewer should accept the author's preference. Otherwise, the choice is dictated by standard principles of software design.
-* If no other rule applies, then the reviewer may ask the author to be consistent with the current codebase, as long as that doesn't worsen the system's overall code health.
+Refactoring is an oppurtunity drive down existing technical debt, optimize, and re-architect the codebase so the code can remain simple, decoupled, easily read, and painlessly scaled. Engineers often complain about old programing languages as if the language is root problem when real problem is old messy code bases.
```
Comment (jmiller-rise8) [inline docs/practice-playbook.md]: typo `oppurtunity`

File: docs/practice-playbook.md [hunk 11]
```diff
@@ -199,49 +288,149 @@ Reviewers should _always_ feel free to leave comments expressing that something
...
 
+Engineers should not shy away from performing large scale refactors (agreed on by the team) that improve the applications performance, ease of use, and deployability. These refactors can range from UI changes to entire revisions of the database schema.
 
-#### Resolving Conflicts
+**Bug**
+Any work being done to correct unexpected behaviors or faults that are inconsistent with the desired coded intent.
 
-In any conflict on a code review, the first step should always be for the developer and reviewer to reach an agreement.
+**NOTE**
+Security is a fundamental part of software developemnt and as such can be characterized to fit in all three of the categories as needed.  In a high compliance environment where stories are created to address security controls from Compliance tool such as SD Elements the team may want a seprate category to cover this work.
```
Comment (jmiller-rise8) [inline docs/practice-playbook.md]: typos

File: docs/practice-playbook.md [hunk 12]
```diff
@@ -199,49 +288,149 @@ Reviewers should _always_ feel free to leave comments expressing that something
...
+
+CI stages should include at a minimum
+- linting
+- unit tests
+- static code analysis
+- dependency scans
+
+
+**Continuous Delivery**
+We believe that any merge to main should be able to deploy to production and main should always be deployable. Merges should be self-contained and not dependant upon another branch.
```
Comment (jmiller-rise8) [inline docs/practice-playbook.md:285]: TODO: Need to address this more. Maybe remove for now?

File: docs/practice-playbook.md [hunk 13]
```diff
@@ -199,49 +288,149 @@ Reviewers should _always_ feel free to leave comments expressing that something
...
+
+
+### Unit
+
+The unit test is designed to test a small, singular component/function/method. The tests are easy to author and maintain and thus often represent the largest portion of tests within the code.
+
+
+### Contract Testing
+
+TDD for micro service architecture, contracts are written on what will be consumed and then consumers and producers are tested against these contracts.  Eliminates the need for test environments that have all services running and at a specific version.
```
Comment (jmiller-rise8) [inline docs/practice-playbook.md]: Maybe generalize a bit more. Micro services are just a single use case

File: docs/practice-playbook.md [hunk 14]
```diff
@@ -199,49 +288,149 @@ Reviewers should _always_ feel free to leave comments expressing that something
...
+
+
+### Integration
+
+The integration test is designed to test between components. A typical example might be integrating with a database or a provided REST service. Integration tests require that you stand up not only your product but also the components with which you integrate. For this reason, they require more time and effort than unit tests. They often times are the second most frequently used test.
+
+
+### End to End {#end-to-end}
```
Comment (jmiller-rise8) [inline docs/practice-playbook.md:301]: Leaning towards Acceptance Tests?

Comment (jmiller-rise8) [reply docs/practice-playbook.md:301]: Add Acceptance Tests?

File: docs/practice-playbook.md [hunk 15]
```diff
@@ -199,49 +288,149 @@ Reviewers should _always_ feel free to leave comments expressing that something
...
+- Observability
+- Alerting
+- Incident response
 
 
 ## Modern Applications
 
 
-### Logging
+### Logging (Operating apps)
```
Comment (jmiller-rise8) [inline docs/practice-playbook.md:312]: @Zojii We haven't visited these topics down

File: docs/practice-playbook.md (hunk 12 above)
Comment (Zojii) [reply docs/practice-playbook.md:285]: good engough for now. I added a comment to re-attack

File: docs/practice-playbook.md (hunk 14 above)
Comment (Zojii) [reply docs/practice-playbook.md:301]: adding todo for acceptance tests

File: docs/practice-playbook.md (hunk 15 above)
Comment (Zojii) [reply docs/practice-playbook.md:312]: catch it. next round

**Review by Zojii — APPROVED** [review]

**Review by dlamberson-rise8 — APPROVED** [review]

**Pull request merged**